stdlib/zipfile.pyi

```diff
@@ -205,20 +219,25 @@ class ZipInfo:
     def FileHeader(self, zip64: bool | None = ...) -> bytes: ...
 
 class _PathOpenProtocol(Protocol):
-    def __call__(self, mode: str = ..., pwd: bytes | None = ..., *, force_zip64: bool = ...) -> IO[bytes]: ...
+    def __call__(self, mode: _ReadWriteMode = ..., pwd: bytes | None = ..., *, force_zip64: bool = ...) -> IO[bytes]: ...
 
 if sys.version_info >= (3, 8):
     class Path:
         @property
         def name(self) -> str: ...
-        @property
-        def parent(self) -> Path: ...  # undocumented
         if sys.version_info >= (3, 10):
             @property
-            def filename(self) -> PathLike[str]: ...
+            def filename(self) -> PathLike[str]: ...  # undocumented
+            @property
+            def parent(self) -> Path | PathLike[str]: ...  # undocumented
+        else:
+            @property
+            def parent(self) -> Path: ...  # undocumented
```

Comment:
For simplicity's sake, I'd recommend to just return `PathLike[str]` from `parent` for all versions.